### DISCUSSION TITLE
Keyboard accessibility

### DISCUSSION THREAD

**Discussion by IeuanWalker** [dotnet/maui#6486] [General]
Hi, really liking the focus on accessibility in MAUI, but read over this doc and found no mention of keyboard accessibility - 
https://docs.microsoft.com/en-us/dotnet/maui/fundamentals/accessibility

Keyboard and screen reader users have different accessibility needs, and over the past year I have found that things that work for screen reader users dont always work for keyboard users (and vice-versa).
https://tink.uk/the-difference-between-keyboard-and-screen-reader-navigation/#:~:text=Keyboard%20users%3A%20people%20who%20don,can't%20see%20the%20content.

So just wondering if the doc is going to be updated with more detail on keyboard navigation.

---

Here are some things of the top of my head

- On android when clicking an element using the screen reader or not, it triggers the normal down/ up events. Whereas clicking an element using a keyboard you have to specifically look for the enter/space key as it wont trigger the normal down/up events - [example](https://github.com/IeuanWalker/Xamarin.Forms.StateButton/blob/e56a3c037ed8dbd7e5235a0987df200aea13810c/Scr/StateButton.Android/StateButtonRenderer.cs#L95)
- XF TapGestureRecognizer doesn't work at all with keyboard. Possible related to the above point
- Important to note that when navigating with just a keyboard only interactive elements should be focusable, where as with screen reader elemets with information or is interactive should be focusable
- To test keyboard navigation on iOS you have to make sure you turn on 'Full keyboard access' (Settings > Accessibility > Keyboard > Full keboard access)

**Comment by kubaflo** [comment]
This should help: https://github.com/dotnet/maui/pull/29649
